File: hang.liu/1299.js
var replaceElements = function(arr) {
    arr.push(-1)
    
    for (var i = arr.length - 2; i > 0; i--) {
        arr[i] = Math.max(arr[i], arr[i + 1])
    }
    
    return arr.slice(1)
};

File: xiang.li/1299.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int greatest = -1;
        int temp = -1;
        for(int i = arr.length - 1;i >= 0;i--){
            greatest = temp > greatest ? temp : greatest;
            temp = arr[i];
            arr[i] = greatest;
        }
        return arr;
    }
}
